extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilder.scala

```diff
@@ -65,49 +65,65 @@ object PrivilegesBuilder {
       if (projectionList.isEmpty) {
-        privilegeObjects += PrivilegeObject(table, plan.output.map(_.name))
+        plan match {
+          case pvm: PermanentViewMarker if pvm.output.isEmpty =>
+            privilegeObjects += PrivilegeObject(table, pvm.catalogTable.schema.map(_.name))
+          case _ =>
+            privilegeObjects += PrivilegeObject(table, plan.output.map(_.name))
+        }
       } else {
         val cols = (projectionList ++ conditionList).flatMap(collectLeaves)
           .filter(plan.outputSet.contains).map(_.name).distinct
         privilegeObjects += PrivilegeObject(table, cols)
       }
     }
 
+    // Should execute this for all plan and skip PermanentViewMaker
+    def buildSubquery(): Unit = {
+      plan transformExpressions {
+        case subquery: SubqueryExpression =>
+          buildQuery(subquery.plan, privilegeObjects, projectionList, conditionList, spark)
+          subquery
+      }
+    }
+
     plan match {
-      case p: Project => buildQuery(p.child, privilegeObjects, p.projectList, conditionList, spark)
+      case p: Project =>
+        buildSubquery()
+        buildQuery(p.child, privilegeObjects, p.projectList, conditionList, spark)
 
       case j: Join =>
         val cols =
           conditionList ++ j.condition.map(expr => collectLeaves(expr)).getOrElse(Nil)
+        buildSubquery()
         buildQuery(j.left, privilegeObjects, projectionList, cols, spark)
         buildQuery(j.right, privilegeObjects, projectionList, cols, spark)
 
       case f: Filter =>
         val cols = conditionList ++ collectLeaves(f.condition)
+        buildSubquery()
         buildQuery(f.child, privilegeObjects, projectionList, cols, spark)
 
       case w: Window =>
         val orderCols = w.orderSpec.flatMap(orderSpec => collectLeaves(orderSpec))
         val partitionCols = w.partitionSpec.flatMap(partitionSpec => collectLeaves(partitionSpec))
         val cols = conditionList ++ orderCols ++ partitionCols
+        buildSubquery()
         buildQuery(w.child, privilegeObjects, projectionList, cols, spark)
 
       case s: Sort =>
         val sortCols = s.order.flatMap(sortOrder => collectLeaves(sortOrder))
         val cols = conditionList ++ sortCols
+        buildSubquery()
         buildQuery(s.child, privilegeObjects, projectionList, cols, spark)
 
       case a: Aggregate =>
         val aggCols =
           (a.aggregateExpressions ++ a.groupingExpressions).flatMap(e => collectLeaves(e))
         val cols = conditionList ++ aggCols
+        buildSubquery()
         buildQuery(a.child, privilegeObjects, projectionList, cols, spark)
 
-      case pvm: PermanentViewMarker =>
-        getScanSpec(pvm).tables(pvm, spark).foreach { table =>
-          privilegeObjects += PrivilegeObject(table, pvm.visitColNames)
-        }
-
       case scan if isKnownScan(scan) && scan.resolved =>
         getScanSpec(scan).tables(scan, spark).foreach(mergeProjection(_, scan))
```

Comment:
Son't need to handle SubqueryExpression for scan node and UnresolvedRelation node